### PR TITLE
[master < MG] Order chunks in `utils::Pool` to speed up deallocation

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/utils/memory.cpp [hunk 1]
```diff
@@ -187,14 +187,19 @@ void *Pool::Allocate() {
   for (unsigned char i = 0U; i < blocks_per_chunk_; ++i) {
     *(data + (i * block_size_)) = i + 1U;
   }
+  Chunk chunk{data, 0, blocks_per_chunk_};
+  // Insert the big block in the sorted position.
+  auto it = std::lower_bound(chunks_.begin(), chunks_.end(), chunk,
+                             [](const auto &a, const auto &b) { return a.data < b.data; });
```
Comment (antoniofilipovic) [inline src/utils/memory.cpp:193]: I have a question here. Isn't it that all Chunks in one Pool have same size of data?

**Review by antoniofilipovic — APPROVED** [review]
Works really well, I am not sure 100% how it works and what exactly happens in the ordering of chunks and deallocating

**Comment by antoniofilipovic** [comment]
@vpavicic this was also merged.

Release note: Speed up a bug on deallocation of resources in Memgraph's custom PoolResource memory allocator

**Comment by vpavicic** [comment]
Performance has been improved by optimizing the deallocation of resources in Memgraph's custom `PoolResource` memory allocator.

@antoniofilipovic shout if not oK :)

**Pull request merged**